lib/puppet/util/windows/adsi.rb

```
@@ -162,6 +162,13 @@ def commit
       begin
         native_user.SetInfo unless native_user.nil?
       rescue Exception => e
+        if e.message =~ /8007089A/m
```

I'll just let the error bubble up if it is not a WIN32 OLE Runtime Error.